flink-yarn/src/test/java/org/apache/flink/yarn/YarnResourceManagerTest.java

```diff
@@ -359,6 +360,32 @@ ContainerStatus createTestingContainerStatus(final ContainerId containerId) {
 		}
 	}
 
+	@Test
+	public void testRunAsyncCausesFatalError() throws Exception {
+		new Context() {{
+			runTest(() -> {
+				// allocate a new container
+				registerSlotRequest(resourceManager, rmServices, resourceProfile1, taskHost);
+				Container testingContainer = createTestingContainer();
+				resourceManager.onContainersAllocated(ImmutableList.of(testingContainer));
+
+				// let client throw exception when requesting a new container
+				testingYarnAMRMClientAsync.setAddContainerRequestConsumer((ignored1, ignored2) -> {
+					throw new RuntimeException("Expected Exception");
+				});
+
+				// complete the container and request a new one
+				ContainerStatus testingContainerStatus = createTestingContainerStatus(testingContainer.getId());
+				resourceManager.onContainersCompleted(Collections.singletonList(testingContainerStatus));
```

Comment:
I think the test passes also w/o these lines. Can we make it explicit where the exception is thrown which is reported via the `TestingFatalErrorHandler`?